client/app/cloud/project/openstack/users/openstack-users.html

```diff
@@ -9,277 +9,57 @@
-
+<oui-datagrid rows="CloudProjectOpenstackUsersCtrl.table.users">
+    <oui-column title="'openstackusers_users_userlist_username_cols' | translate" property="username" sortable></oui-column>
+    <oui-column title="'openstackusers_users_userlist_description_cols' | translate" property="description" sortable></oui-column>
+    <oui-column title="'openstackusers_users_userlist_password_cols' | translate" property="id">
+        <div data-ng-class="{
+             'opacity2-leave' : CloudProjectOpenstackUsersCtrl.toggle.userDeleteId === $row.id,
+             'opacity0-leave' : CloudProjectOpenstackUsersCtrl.toggle.userGenerateTokenId === $row.id,
+             'opacity2-enter' : CloudProjectOpenstackUsersCtrl.toggle.userDeleteId !== $row.id && CloudProjectOpenstackUsersCtrl.toggle.userGenerateTokenId !== $row.id}">
+            <p class="no-space" data-ng-show="CloudProjectOpenstackUsersCtrl.getPassword($row)">{{ CloudProjectOpenstackUsersCtrl.getPassword($row) }}</p>
+            <span data-ng-show="!CloudProjectOpenstackUsersCtrl.getPassword($row)">
+                <i class="fa fa-circle"></i>
+                <i class="fa fa-circle"></i>
+                <i class="fa fa-circle"></i>
+                <i class="fa fa-circle"></i>
+                <i class="fa fa-circle"></i>
+                <i class="fa fa-circle"></i>
+                <i class="fa fa-circle"></i>
+            </span>
+            <button title="{{ 'openstackusers_users_userlist_regenerate_password' | translate }}"
+                    class="no-style"
+                    data-ng-if="!CloudProjectOpenstackUsersCtrl.getPassword($row)"
+                    data-ng-disabled=" $row.status === 'disabled'"
+                    data-ng-click="CloudProjectOpenstackUsersCtrl.regeneratePassword($row)"
+                    write-right-required>
+                <i data-ng-hide="CloudProjectOpenstackUsersCtrl.loaders.regeneratePassword === $row.id" class="fa fa-refresh"></i>
+                <oui-spinner size="s" data-ng-show="CloudProjectOpenstackUsersCtrl.loaders.regeneratePassword === $row.id"></oui-spinner>
+            </button>
+        </div>
+    </oui-column>
+
+    <oui-column title="'openstackusers_users_userlist_status_cols' | translate" property="status" sortable>
+        <span class="label"
+              data-ng-class="{
+              'opacity2-leave'    : CloudProjectOpenstackUsersCtrl.toggle.userDeleteId === $row.id,
+              'opacity2-enter'    : CloudProjectOpenstackUsersCtrl.toggle.userDeleteId !== $row.id,
+              'label-danger'      : $row.status === 'deleted' || $row.status === 'deleting' || $row.status === 'disabled',
+              'label-info'        : $row.status === 'creating' || $row.status === 'updating',
+              'label-success'     : $row.status === 'ok'
+              }">
+            {{ $row.status }}
+        </span>
+    </oui-column>
+    <oui-action-menu data-align="end" compact>
```

Comment:
we have to decide whether we use data-xyz or not.  You use data-align on line 55 and when on line 56-57 it is gone. ;)

Reply:
yes, we need to use prefixes...